Import the latest cookie prefixes and cookie-store special names WPTs

yoavweiss opened the pull request:
#### 8afd08557be462ffe1b3cf06cd39c806dcf37afc
<pre>
Import the latest cookie prefixes and cookie-store special names WPTs
<a href="https://bugs.webkit.org/show_bug.cgi?id=295007">https://bugs.webkit.org/show_bug.cgi?id=295007</a>

Reviewed by Anne van Kesteren.

This imports the latest WPTs for both the cookieStore API and cookies prefixes.
Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/5aadede8c5042de90d80a9875c018679b89f70ab">https://github.com/web-platform-tests/wpt/commit/5aadede8c5042de90d80a9875c018679b89f70ab</a>

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_special_names.https.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_special_names.https.any.js:
(string_appeared_here.forEach.prefix.promise_test.async testCase):
(promise_test.async testCase):
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_special_names.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookies/prefix/__HostHttp.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/cookies/prefix/__HostHttp.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/cookies/prefix/__Http.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/cookies/prefix/__Http.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/cookies/prefix/__host.document-cookie-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookies/prefix/__host.document-cookie.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookies/prefix/__host.header-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookies/prefix/__host.header.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookies/prefix/__secure.document-cookie-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookies/prefix/__secure.document-cookie.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookies/prefix/__secure.header-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookies/prefix/__secure.header.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookies/prefix/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/cookies/resources/cookie-helper.sub.js:
(set_prefixed_cookie_via_dom_test): Ensure the tests have a consistent output.
(set_prefixed_cookie_via_http_test): Ensure the tests have a consistent output.
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/cookies/prefix/__HostHttp.https-expected.txt: Added.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/cookies/prefix/__HostHttp.https-expected.txt: Added.

Canonical link: <a href="https://commits.webkit.org/297126@main">https://commits.webkit.org/297126@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9441bc0f993db8381aa9e6fb779bbee9f7cd4602

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/110560 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/30219 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/20652 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/116586 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/60827 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/30898 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/38807 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/84077 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/113508 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/24671 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/99554 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/64518 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/24034 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/17694 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/60381 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/94056 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/17753 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/119376 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/37600 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/27918 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/93047 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/37974 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/95825 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/92865 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23671 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/37874 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/15614 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/33573 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/37495 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/37158 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/40497 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/38866 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk reviewed (commented):
This seems okay, modulo EWS failures, but maybe we can rename the tests away from `.tentative` upstream first? I'll land the specification change today or tomorrow.

annevk reviewed (commented):
Did you check whether those tests with inconsistent output are currently disabled?

annevk commented on LayoutTests/TestExpectations:
```diff
@@ -869,13 +869,6 @@ imported/w3c/web-platform-tests/workers/interfaces/WorkerGlobalScope/onerror/mes
-imported/w3c/web-platform-tests/cookies/prefix/__secure.document-cookie.html [ Pass Failure ]
-imported/w3c/web-platform-tests/cookies/prefix/__secure.document-cookie.https.html [ Pass Failure ]
-imported/w3c/web-platform-tests/cookies/prefix/__secure.header.html [ Pass Failure ]
 imported/w3c/web-platform-tests/cookies/prefix/document-cookie.non-secure.html [ Pass Failure ]
```
Would be nice to fix this one too.

yoavweiss replied:
https://github.com/web-platform-tests/wpt/pull/53653

annevk reviewed (approved):
Looks good modulo bots.

yoavweiss committed:
Import the latest cookie prefixes and cookie-store special names WPTs
https://bugs.webkit.org/show_bug.cgi?id=295007

Reviewed by Anne van Kesteren.

This imports the latest WPTs for both the cookieStore API and cookies prefixes.
Upstream commit: https://github.com/web-platform-tests/wpt/commit/5aadede8c5042de90d80a9875c018679b89f70ab

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_special_names.https.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_special_names.https.any.js:
(string_appeared_here.forEach.prefix.promise_test.async testCase):
(promise_test.async testCase):
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_special_names.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookies/prefix/__HostHttp.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/cookies/prefix/__HostHttp.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/cookies/prefix/__Http.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/cookies/prefix/__Http.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/cookies/prefix/__host.document-cookie-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookies/prefix/__host.document-cookie.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookies/prefix/__host.header-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookies/prefix/__host.header.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookies/prefix/__secure.document-cookie-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookies/prefix/__secure.document-cookie.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookies/prefix/__secure.header-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookies/prefix/__secure.header.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookies/prefix/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/cookies/resources/cookie-helper.sub.js:
(set_prefixed_cookie_via_dom_test): Ensure the tests have a consistent output.
(set_prefixed_cookie_via_http_test): Ensure the tests have a consistent output.
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/cookies/prefix/__HostHttp.https-expected.txt: Added.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/cookies/prefix/__HostHttp.https-expected.txt: Added.

Canonical link: https://commits.webkit.org/297126@main
